core/src/main/java/contacts/core/entities/DataEntity.kt

```diff
@@ -124,9 +124,19 @@ interface CommonDataEntity : Entity {
 }
 
 /**
- * A [CommonDataEntity] that is mutable, allowing the [primaryValue] to be mutated among others.
+ * A [DataEntity] that is immutable.
+ *
+ * ## Dev note
+ *
+ * We are explicitly prepending "Immutable" to the interface name to prevent naming conflicts with
+ * [contacts.core.data.Data].
```

Comment:
I actually initially named this as just `Data` but then I realized that I'm already using that name for the `Data` API 🤦. Whatever, this is good too. Some would say it is actually better as it is more verbose!